day_two_guide/topics/pruning_images_and_containers.adoc

```diff
@@ -0,0 +1,432 @@
+Dec 07 10:35:34 app-node-1.example.com atomic-openshift-node[3390]: I1207 10:35:34.264344    3390 image_gc_manager.go:270] *[imageGCManager]: Disk usage on "/dev/vda1" (/) is at 77% which is over the high threshold (75%).*
+Dec 07 10:40:37 app-node-1.example.com atomic-openshift-node[3390]: I1207 10:40:37.287970    3390 image_gc_manager.go:335] *[imageGCManager]: Removing image "sha256:0a2c8543c09d4ed7a121f3bf2fc0a76016b183aab3d28a471492e07fcd9e0457" to free 540485833 bytes*
+----
+
+Observe error messages if the pruning process is failing such as:
+
+----
+Dec 11 11:26:02 app-node-1.example.com atomic-openshift-node[3390]: E1211 11:26:02.271772    3390 kubelet.go:1170] Image garbage collection failed: wanted to free 2317889740, but freed 0 space with errors in image deletion: [rpc error: code = 2 desc = Error response from daemon: {"message":"conflict: unable to delete 161690d448ba (cannot be forced) - image has dependent child images"}, rpc error: code = 2 desc = Error response from daemon: {"message":"conflict: unable to delete 3a57255d6707 (must be forced) - image is referenced in one or more repositories"}]
+----
+
+The image garbage collection process doesn't remove images pulled manually using
+`docker pull` or `docker run` related tasks, so if needed, the images
+can be deleted using `docker rmi` as:
+
+[subs=+quotes]
+----
+$ sudo docker rmi *<my_manually_pulled_image:tag>*
+----
+
+In the event of a failure of garbage collection, manual deletion of unused images is required:
+
+----
+$ sudo docker rmi $(sudo docker images -f "dangling=true" -q)
+----
+
+WARNING: While not recommended, a more aggressive clean up can be performed
+by attempting to delete all the images. 
+`docker` protects images it is using; this can be performed as `sudo docker rmi $(sudo docker images -q)`
+
+==== Container Garbage Collection
+Old pods are not removed immediately as the user may want to perform some tasks
+such as seeing the logs of the failed pod or the failed build using
+`oc logs -p`. This leads to stopped containers living in the
+nodes. To see the unused containers, perform the following step in any node:
+
+----
+$ diff -w -b <(sudo docker ps) <(sudo docker ps -a)
+----
+
+{rhocp} performs a garbage collection process for dead containers that can
+be tweaked depending on three variables:
+
+* `minimum-container-ttl-duration` is the time since the pod is dead until it can be chosen to be deleted. Default value is set to 1 minute (1m). 0 to disable it.
+
+* `maximum-dead-containers-per-container` is the maximum number of dead containers every single pod (UID, container name) pair is allowed to have. Default value is set to 2. -1 to disable it. For instance, the pod named "myawesomeapp" when instantiated become "myawesomeapp-1-6mp91" that contains a container from "myregistry/myawesomeapp:tag" image and with the default setting, two instances of that container are allowed to be present in the host.
```

Comment:
'instantiated become" should be 'instantiated becomes'

'that contains' should be 'and contains'

'contains a container from' should be 'contains a container from the'